extension/chrome/elements/compose-modules/formatters/general-mail-formatter.ts

```diff
@@ -80,4 +89,24 @@ export class GeneralMailFormatter {
     // throws ComposerResetBtnTrigger when user closes pass phrase dialog without entering
     return await view.storageModule.decryptSenderKey(parsedSenderPrv);
   };
+
+  private static isSenderInKeyUserIds = async (view: ComposeView, signingKey: ParsedKeyInfo): Promise<boolean> => {
+    const senderEmail = view.senderModule.getSender();
+    if (senderEmail !== view.acctEmail && signingKey.keyInfo.emails) {
+      const storage = await AcctStore.get(view.acctEmail, ['sendAs']);
+      const emailAliases = Object.keys(storage.sendAs || {});
+      const senderAliasIndex = emailAliases.indexOf(view.acctEmail);
+      if (senderAliasIndex !== -1) {
+        emailAliases.splice(senderAliasIndex, 1);
+      }
+      const emailsInSigningKey = signingKey.keyInfo.emails || [];
+      let filteredEmailsInSigningKey: string[] = [];
+      for (const email of emailsInSigningKey) {
+        filteredEmailsInSigningKey.push(email.replace(/(.+)(?=\+).*(?=@)/, '$1'));
+      }
+      filteredEmailsInSigningKey = filteredEmailsInSigningKey.filter((value, index) => filteredEmailsInSigningKey.indexOf(value) === index);
+      return emailAliases.every(email => filteredEmailsInSigningKey.includes(email));
```

Comment:
As I understand, currently you fetch all user aliases from `AcctStore`, then get all emails from signing key and check if signing key emails contain all users aliases?

But I think we don't need to check for all user aliases, as user uses only 1 alias for sending message, so correct check should be just:
- get all emails from signing key
- check if these emails include `senderEmail`

Probably it should be enough.

P.S. Can you please explain what filtering you perform here (as it's difficult to understand without additional comments, feel free to add them to the code):
```ts
let filteredEmailsInSigningKey: string[] = [];
for (const email of emailsInSigningKey) {
    filteredEmailsInSigningKey.push(email.replace(/(.+)(?=\+).*(?=@)/, '$1'));
}
filteredEmailsInSigningKey = filteredEmailsInSigningKey.filter((value, index) => filteredEmailsInSigningKey.indexOf(value) === index);
```

Reply:
Yes, Roma, my bad. I have updated it and re-considered the aliases with filters "+" and sanitize them for better matching.